hotel_reservation/report/hotel_reservation_report.py

```diff
@@ -0,0 +1,269 @@
+            "get_room_nos": _get_room_nos,
+            "get_checkin": get_checkin,
+        }
+
+
+class ReportTestCheckout(models.AbstractModel):
+    _name = "report.hotel_reservation.report_checkout_qweb"
+    _description = "Auxiliar to get the check out report"
+
+    def _get_room_type(self, date_start, date_end):
+        reservation_obj = self.env["hotel.reservation"]
+        res = reservation_obj.search(
+            [("checkout", ">=", date_start), ("checkout", "<=", date_end)]
+        )
+        return res
+
+    def _get_room_nos(self, date_start, date_end):
+        reservation_obj = self.env["hotel.reservation"]
+        res = reservation_obj.search(
+            [("checkout", ">=", date_start), ("checkout", "<=", date_end)]
+        )
+        return res
+
+    def _get_checkout(self, date_start, date_end):
+        reservation_obj = self.env["hotel.reservation"]
+        res = reservation_obj.search(
+            [("checkout", ">=", date_start), ("checkout", "<=", date_end)]
+        )
+        return res
+
+    @api.model
+    def _get_report_values(self, docids, data):
+        self.model = self.env.context.get("active_model")
+        if data is None:
+            data = {}
+        if not docids:
+            docids = data["form"].get("docids")
+        folio_profile = self.env["hotel.reservation"].browse(docids)
+        date_start = data.get("date_start", fields.Date.today())
+        date_end = data["form"].get(
+            "date_end",
+            str(datetime.now() + relativedelta(months=+1, day=1, days=-1))[
+                :10
+            ],
+        )
+        rm_act = self.with_context(data["form"].get("used_context", {}))
+        _get_room_type = rm_act._get_room_type(date_start, date_end)
+        _get_room_nos = rm_act._get_room_nos(date_start, date_end)
+        _get_checkout = rm_act._get_checkout(date_start, date_end)
+        return {
+            "doc_ids": docids,
+            "doc_model": self.model,
+            "data": data["form"],
+            "docs": folio_profile,
+            "time": time,
+            "get_room_type": _get_room_type,
+            "get_room_nos": _get_room_nos,
+            "get_checkout": _get_checkout,
+        }
+
+
+class ReportTestMaxroom(models.AbstractModel):
+    _name = "report.hotel_reservation.report_maxroom_qweb"
+    _description = "Auxiliar to get the room report"
+
+    def _get_room_type(self, date_start, date_end):
+        reservation_obj = self.env["hotel.reservation"]
+        tids = reservation_obj.search(
+            [("checkin", ">=", date_start), ("checkout", "<=", date_end)]
+        )
+        res = reservation_obj.browse(tids)
+        return res
+
+    def _get_room_nos(self, date_start, date_end):
+        reservation_obj = self.env["hotel.reservation"]
+        tids = reservation_obj.search(
+            [("checkin", ">=", date_start), ("checkout", "<=", date_end)]
+        )
+        res = reservation_obj.browse(tids)
```

Comment:
`tids` is itself as browse records, please remove it

Reply:
Done